### DISCUSSION TITLE
npm `package-lock.json` updated registry host configured in npmrc

### DISCUSSION THREAD

**Discussion by larhauga** [renovatebot/renovate#18002] [Request Help]
we are running a nexus repository cache for npm packages, while clients use a different host against the nexus cache. The client reachable host address is persisted in the `package-lock.json` files.

So we wanted renovate to use the nexus proxy to do all lookup operations, and added this configuration:
```
"npmrc": "registry=https://nexus.nexus.svc.cluster.local/repository/npm-proxy/"
```

But now the `package-lock.json` file gets updated with the new address.

```
"@material-ui/icons": {
...
    "resolved": "https://nexus.nexus.svc.cluster.local/repository/npm-proxy/@material-ui/icons/-/icons-xxx.tgz",
...},
```

Is it possible to have renovate do all operations against nexus, without having it update the resolved host in the `package-lock.json` file?
Appreciate any pointers :)

**Comment by viceice** [comment]
no, as it's the npm manager which stores those urls.
